src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,203 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        final Node<T> link = last;
+        final Node<T> node = new Node<>(link, value, null);
+        last = node;
+        if (link == null) {
+            first = node;
+        } else {
+            link.next = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            linkBefore(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            return unlinkFirst();
+        } else if (index == size - 1) {
+            return unlinkLast();
+        } else {
+            return unlink(index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (getIndex(object) != -1) {
+            remove(getIndex(object));
+            return true;
+        }
```

Comment:
lets iterate here and unlink node if you find a value
because when you are looking for an index for node you are iterating the whole list for the first time
and you would iterate again when removing it by index